File: mergeIntervals.py


class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        ans = []
        sortedIntervals = sorted(intervals, key = itemgetter(0))
        
        ans.append(sortedIntervals[0])
        
        for i in range(1, len(intervals)):
            if(ans[-1][1] < sortedIntervals[i][0]):
                ans.append(sortedIntervals[i])
            else:
                ans[-1][1] = max(ans[-1][1], sortedIntervals[i][1])
        
        return ans

File: mergeIntervals.cpp
class Solution {
public:
    static bool sortByLeft(vector<int>& a, vector<int>& b) {
        return a[0]<b[0];
    }
    vector<vector<int>> merge(vector<vector<int>>& inter) {
        sort(inter.begin(), inter.end(), sortByLeft);
        int l = inter[0][0];
        int r = inter[0][1];
        vector<vector<int>> ans;
        for(int i=1; i<inter.size(); i++) {
            if(inter[i][0] <= r && inter[i][1] > r) r = inter[i][1];
            else if(inter[i][0] <= r) continue;
            else {
                ans.push_back({l, r});
                l = inter[i][0];
                r = inter[i][1];
            }
        }
        ans.push_back({l, r});
        return ans;
    }
};